eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -292,18 +311,19 @@ object MultiPartHandler {
 
     // @formatter:off
     sealed trait Command
-    case class CreateInvoice(replyTo: ActorRef, receivePayment: ReceivePayment) extends Command
+    case class CreateInvoice(replyTo: typed.ActorRef[Try[Invoice]], receivePayment: ReceivePayment) extends Command
+    case class WrappedInvoice(invoice: Try[Bolt12Invoice]) extends Command
     // @formatter:on
 
     def apply(nodeParams: NodeParams): Behavior[Command] = {
       Behaviors.setup { context =>
-        Behaviors.receiveMessage {
+        Behaviors.receiveMessagePartial {
           case CreateInvoice(replyTo, receivePayment) =>
             val paymentPreimage = receivePayment.paymentPreimage_opt.getOrElse(randomBytes32())
             val paymentHash = Crypto.sha256(paymentPreimage)
             receivePayment match {
               case r: ReceiveStandardPayment =>
-                Try {
```

Comment:
But these `require` are here to catch _developer_ bugs, not _runtime_ issues: none of them can trigger outside of unit tests, since we don't let node operators disable `Features.PaymentSecret` and we make sure that we don't pass in features that contain `Features.RouteBlinding`. None of these `require` can be influenced by outside data, so we shouldn't treat this as a `Try` (otherwise we would have made that `Bolt11Invoice` return a `Try[Bolt11Invoice]`).

Reply:
I gave it a try in https://github.com/ACINQ/eclair/commit/07c1a84663ebf88aea3057fa82112b367408c95a and it's looking better IMO